### ISSUE TITLE
[QUESTION] Limitation of TensorRT I/O format when using a Cutlass convolution

### ISSUE THREAD

**Issue by remi-or-dtk** [NVIDIA/TensorRT#3192]
Hello,

This is not so much an issue with a particuliar TensorRT version as much  a general compatibility issue between two NVIDIA software. 
According to the [documentation of TensorRT](https://docs.nvidia.com/deeplearning/tensorrt/developer-guide/index.html#reformat-free-network-tensors) the only supported I/O format for INT8 data types are either kLinear or kCHWx. As I/O also applies to passed passed to and from plugins, this means an INT8 plugin should have this data format to avoid reformatting. 
Now, I mainly want to leverage Cutlass in my plugin, and they have confirmed to me that their ImplicitGemm convolutions [only support channels as the leading dimension](https://github.com/NVIDIA/cutlass/issues/978#issuecomment-1662288399). Plus, its still unclear that I can use bias when using an interleaved format, so my only real option right now is to use NHWC.
But this is unfortunate, as both these products are NVIDIA-made. 

Is there any way to pass INT8+kHWC data to a plugin in TensorRT? I'm open to any tricks you might have. Thanks.

**Comment by zerollzeng** [comment]
Yes this is a known issue, we only support data format list in https://docs.nvidia.com/deeplearning/tensorrt/developer-guide/index.html#ipluginv2

@samurdhikaru @nvpohanh @ttyio Do we have plan to support all the data format for plugin in the future?

**Comment by nvpohanh** [comment]
We currently don't have the plan to support NHWC INT8 plugin I/O yet. Could you briefly describe what this model is for and why using cutlass is required? This will help us triage this feature request. Thanks

**Comment by remi-or-dtk** [comment]
Sure, this is for quantized models (convolution-based and llms) which uses a quantization scheme that adds some operations before/after weighted layers. The reason I need cutlass is to leverage its speed for GEMMs (and thus convolutions) to make the memory-speed tradeoff viable. Thanks!

**Comment by gujiewen** [comment]
> We currently don't have the plan to support NHWC INT8 plugin I/O yet. Could you briefly describe what this model is for and why using cutlass is required? This will help us triage this feature request. Thanks

any updates or support plans now? I have the same problem.

**Comment by nvpohanh** [comment]
Thanks for describing the use case. we will discuss this internally and will let you know if this is planned for any future TRT version.
